lib/readline.js

```diff
@@ -481,6 +482,42 @@ Interface.prototype._insertString = function(c) {
     // A hack to get the line refreshed if it's needed
     this._moveCursor(0);
   }
+  // Emit current line for generating preview
+  this.emit('buffer', this.line);
+};
+
+// Append eager eval result to the
+// Current line
+Interface.prototype._appendPreview = function(result) {
+  this.previewResult = `\u001b[90m // ${result}\u001b[39m`;
+  const line = `${this._prompt}${this.line} //${result}`;
+  const columns = this.output.columns;
+  const hasColors = this.output.hasColors();
+  const s = hasColors ?
+    `${this._prompt}${this.line}${this.previewResult}` : line;
```

Comment:
I personally would keep the preview, even if no colors are supported. The comment should be indication enough.

If we do not want to have a preview in case the colors are deactivated, we should not start the `eagerSession` in the REPL. That way no code is run unnecessary.

And is it correct that `this._prompt` is necessary in one case but not in the other?

Reply:
Hmmm, I'm planning to send the output to this function and then use `util.inspect` within the function to customize our output. Do you think that would be a better approach? Because anyways I need to call `util.inspect` in my REPL session.